modules/improvedigitalBidAdapter.js

```diff
@@ -0,0 +1,408 @@
+              continue;
+            }
+            if (bidObject.errorCode && bidObject.errorCode !== 0) {
+              bid = bidfactory.createBid(CONSTANTS.STATUS.NO_BID);
+              bid.bidderCode = IMPROVE_DIGITAL_BIDDER_CODE;
+              bidmanager.addBidResponse(bidRequest.placementCode, bid);
+              continue;
+            }
+            if (!bidObject.adm || bidObject.adm === null || typeof bidObject.adm !== 'string') {
+              bid = bidfactory.createBid(CONSTANTS.STATUS.NO_BID);
+              bid.bidderCode = IMPROVE_DIGITAL_BIDDER_CODE;
+              bidmanager.addBidResponse(bidRequest.placementCode, bid);
+              continue;
+            }
+
+            bid = bidfactory.createBid(CONSTANTS.STATUS.GOOD);
+
+            var syncString = '';
+            var syncArray = (bidObject.sync && bidObject.sync.length > 0) ? bidObject.sync : [];
+
+            for (var syncCounter = 0; syncCounter < syncArray.length; syncCounter++) {
+              syncString += (syncString === '') ? 'document.writeln(\"' : '';
+              var syncInd = syncArray[syncCounter];
+              syncInd = syncInd.replace(/\//g, '\\\/');
+              syncString += '<img src=\\\"' + syncInd + '\\\"\/>';
+            }
+            syncString += (syncString === '') ? '' : '\")';
+
+            var nurl = '';
+            if (bidObject.nurl && bidObject.nurl.length > 0) {
+              nurl = '<img src=\"' + bidObject.nurl + '\" width=\"0\" height=\"0\" style=\"display:none\">';
+            }
+            bid.ad = nurl + '<script>' + bidObject.adm + syncString + '</script>';
+            bid.bidderCode = IMPROVE_DIGITAL_BIDDER_CODE;
+            bid.cpm = parseFloat(bidObject.price);
+            bid.width = bidObject.w;
+            bid.height = bidObject.h;
+
+            bidmanager.addBidResponse(bidRequest.placementCode, bid);
+          }
+        }
+      }
+    }
+  };
+
+  baseAdapter.callBids = function (params) {
+    // params will contain an array
+    var bidRequests = params.bids || [];
+    var loc = utils.getTopWindowLocation();
+    var requestParameters = {
+      singleRequestMode: false,
+      httpRequestType: this.idClient.CONSTANTS.HTTP_REQUEST_TYPE.GET,
+      callback: CALLBACK_FUNCTION,
+      secure: (loc.protocol === 'https:') ? 1 : 0,
+      libVersion: this.LIB_VERSION
+    };
+
+    var normalizedBids = [];
+    for (var bidCounter = 0; bidCounter < bidRequests.length; bidCounter++) {
+      var normalizedBidRequest = this.getNormalizedBidRequest(bidRequests[bidCounter]);
+      normalizedBids.push(normalizedBidRequest);
+      if (bidRequests[bidCounter].params && bidRequests[bidCounter].params.singleRequest) {
+        requestParameters.singleRequestMode = true;
+      }
+    }
+
+    var request = this.idClient.createRequest(
+      normalizedBids, // requestObject
+      requestParameters
+    );
+
+    if (request.errors && request.errors.length > 0) {
+      console.log('ID WARNING 0x01');
```

Comment:
Changed the logging to utils.logError.

0 and 1 instead of true or false is a throwback to our internal protocol